### PR TITLE
prover(perf): revisit some bottlenecks on small fields branch

### PR DESCRIPTION
**Pull request by gbotrel**
```
benchmark                                                   old ns/op       new ns/op       delta
BenchmarkCompilerWithSelfRecursion/realistic-segment-96     17504200026     11393877188     -34.91%

benchmark                                                   old allocs     new allocs     delta
BenchmarkCompilerWithSelfRecursion/realistic-segment-96     380613884      3541080        -99.07%

benchmark                                                   old bytes        new bytes       delta
BenchmarkCompilerWithSelfRecursion/realistic-segment-96     115659164560     71497481272     -38.18%
```

And compared to last head:
```
benchmark                                                   old ns/op       new ns/op       delta
BenchmarkCompilerWithSelfRecursion/realistic-segment-96     14469069305     11393877188     -21.25%

benchmark                                                   old allocs     new allocs     delta
BenchmarkCompilerWithSelfRecursion/realistic-segment-96     3480241        3541080        +1.75%

benchmark                                                   old bytes       new bytes       delta
BenchmarkCompilerWithSelfRecursion/realistic-segment-96     81168375664     71497481272     -11.91%
```

That's on a `r7a` I suspect we can be <10s on a `r8a`. 

---

> [!NOTE]
> Major performance refactors: new compiled symbolic VM, faster SMT/Merkle tree build, optimized MPTS quotient and lookup assignments, and unified mixed-field Lagrange evaluation with broad parallelism/alloc reductions.
> 
> - **Performance/Engine**:
>   - Introduce compiled VM evaluator for `symbolic.ExpressionBoard` (new bytecode+VM), replacing old evaluator; add extensive tests and edge-case checks.
> - **MPTS / Quotient**:
>   - Optimize quotient accumulation: group polys by points, vectorized combines, use `PolyPositionInQuery`, arena buffers, and parallel row-wise accumulation.
>   - Random/evaluation paths now use `smartvectors_mixed.BatchEvaluateLagrange`.
> - **Lookups (logderivativesum)**:
>   - Rework `MAssignmentTask.Run` to a partitioned (radix) hash-join with bucketed T/S, handling filters/multiplicity; parallelized; Z task uses `ParBatchInvert` and avoids copies.
> - **SMT/Merkle/Vortex**:
>   - Add `NewTree(leaves)` with cache-friendly parallel subtree build; replace `BuildComplete` usages; add tests and benchmark.
> - **Field/Math utils**:
>   - Add `smartvectors_mixed.BatchEvaluateLagrange` with custom lagrange-basis; remove old batch-eval from `smartvectors`.
>   - Improve `Regular.WriteInSliceExt`, parallelize `vectorext.PowerVec`.
>   - Add `fext.BatchInvertInto` and make `ParBatchInvert` GOMAXPROCS-aware.
> - **Global constraints**:
>   - Evaluation uses mixed-field batch Lagrange; quotient FFT handling refactored; profiling tweaks.
> - **Misc**:
>   - `FiatShamir.UpdateExt` zero-check and `unsafe` slice cast; increase parallel goroutine heuristic; minor FFT task tweaks; poseidon2/split adjustments.
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit 4004c90fa535d04f1901d21d0f783f465e52170d. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>

### CODE REVIEW

File: prover/maths/common/smartvectors_mixed/polynomial.go [hunk 1]
```diff
@@ -69,3 +65,90 @@ func BatchEvaluateLagrange(vs []sv.SmartVector, x fext.Element, oncoset ...bool)
 
 	return results
 }
+
+// TODO taken from gnark-crypto, tuned and adapted for our use case
+// computeLagrangeBasisAtX computes (Lᵢ(x))_{i<n} and numerator for Lagrange basis evaluation
+func computeLagrangeBasisAtX(n int, x fext.Element) extensions.Vector {
+
+	// 1. Compute common factor: (x^n - 1) / n
```
Comment (gbotrel) [inline prover/maths/common/smartvectors_mixed/polynomial.go:73]: @ThomasPiellard can you review this part I vibe coded it with Gemini :)
